hack/mock-unittest-data/gomod.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+import json
+import subprocess
+from contextlib import contextmanager
+from functools import partial
+from itertools import chain
+from os import environ
+from pathlib import Path
+from shutil import copyfile
+from shutil import rmtree as delete_directory_with_contents
+from sys import argv
+from tempfile import gettempdir
+
+from yarn import print_banner
+
+
+def mkdir_p(dirname):
+    return Path(dirname).mkdir(exist_ok=True, parents=True)
+
+
+def generate_auxiliary_dirs(mocked_data_dir):
+    mkdir_p(f"{mocked_data_dir}/non-vendored")
+    mkdir_p(f"{mocked_data_dir}/vendored")
+    mkdir_p(f"{mocked_data_dir}/workspaces")
+    mocked_data_dir_abspath = Path(f"{mocked_data_dir}").absolute()
+
+    tmpdir = Path(gettempdir())
+
+    return mocked_data_dir_abspath, tmpdir
+
+
+@contextmanager
+def pandemonium_dir(pandemonium_root):
+    pandemo_dir = pandemonium_root / "gomod-pandemonium"
+    try:
+        yield pandemo_dir
+    finally:
+        delete_directory_with_contents(pandemo_dir)
```

Comment:
This looks very unnecessary since you're basically just doing:
```
with tempfile.TemporaryDirectory() as  tmpdir:
  x = Path(tmpdir.name) / "gomod-pandemonium" . mkdir()
```

I would argue that this helper context manager not only doesn't improve readabilty, it also goes against some commonly used pattern (tempfile context manager), reimplementing what you already get for free with `tempfile`, please drop this.